app/models/mute.rb

```diff
@@ -0,0 +1,37 @@
+class Mute < ApplicationRecord
+  include MuteHelper
+
+  belongs_to :muter, class_name: "User"
+  belongs_to :muted, class_name: "User"
+
+  validates :muter, :muted, presence: true
+  validates :muted_id, uniqueness: { scope: :muter_id }
+
+  validate :check_self
+  validate :check_official, if: :muted
+  validate :check_mute_limit
```

Comment:
Can you add a test for this? You can temporarily lower the limit; no need to actually make a thousand muted users.

https://github.com/otwcode/otwarchive/blob/09d8ba39c8ec82c070a774bfd4ea479c2ee20c3c/features/step_definitions/kudos_steps.rb#L16-L18